src/backends/plonky2/primitives/ec/curve.rs

```diff
@@ -606,7 +716,19 @@ impl CircuitBuilderElliptic for CircuitBuilder<GoldilocksField, 2> {
         inputs.extend_from_slice(&p1.u.components);
         inputs.extend_from_slice(&p2.x.components);
         inputs.extend_from_slice(&p2.u.components);
-        let outputs = ECAddHomogOffset::apply(self, &inputs);
+        let gate = ECAddHomogOffsetGate::new_from_config(&self.config);
+        let (gate, i) = self.find_slot(gate, &[], &[]);
+        let wires_a0 = ECAddHomogOffsetGate::wires_ith_addend_0(i)
+            .map(|i| Target::wire(gate, i))
+            .collect::<Vec<_>>();
+        let wires_a1 = ECAddHomogOffsetGate::wires_ith_addend_1(i)
+            .map(|i| Target::wire(gate, i))
+            .collect::<Vec<_>>();
+        let outputs = ECAddHomogOffsetGate::wires_ith_output(i)
+            .map(|i| Target::wire(gate, i))
+            .collect::<Vec<_>>();
+        zip_eq(inputs, [wires_a0, wires_a1].concat()).for_each(|(i, w)| self.connect(i, w));
```

Comment:
Did you consider adding an `apply` method to `ECAddHomogOffsetGate` that performs these connections of inputs and returns the outputs?